spark/src/main/java/io/ddf/spark/SparkDDFManager.java

```diff
@@ -31,58 +37,170 @@
 import java.security.SecureRandom;
-import java.util.*;
-
-import org.apache.spark.sql.types.StructType;
-
-import io.ddf.s3.S3DDF;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+
+import scala.Tuple2;
+import scala.Tuple3;
+import scala.collection.JavaConversions;
+import scala.collection.convert.WrapAsJava;
 
 /**
  * An Apache-Spark-based implementation of DDFManager
  */
 public class SparkDDFManager extends DDFManager {
 
-  @Override
-  public String getEngine() {
-    return "spark";
-  }
-
-
   private static final String DEFAULT_SPARK_APPNAME = "DDFClient";
   private static final String DEFAULT_SPARK_MASTER = "local[4]";
+  private static final String[][] SPARK_ENV_VARS = new String[][]{
+      // @formatter:off
+      {"SPARK_APPNAME", "spark.appname"},
+      {"SPARK_MASTER", "spark.master"},
+      {"SPARK_HOME", "spark.home"},
+      {"SPARK_SERIALIZER", "spark.kryo.registrator"},
+      {"HIVE_HOME", "hive.home"},
+      {"HADOOP_HOME", "hadoop.home"},
+      {"DDFSPARK_JAR", "ddfspark.jar"}
+      // @formatter:on
+  };
+  private SparkContext mSparkContext;
+  private JavaSparkContext mJavaSparkContext;
+  private HiveContext mHiveContext;
+  /**
+   * Also calls setSparkContext() to the same sharkContext
+   *
+   * @param sharkContext
+   */
+  //  private void setSharkContext(SharkContext sharkContext) {
+  //    this.mSharkContext = sharkContext;
+  //    this.setSparkContext(sharkContext);
+  //  }
+
+
+  private Map<String, String> mSparkContextParams;
 
   public SparkDDFManager(SparkContext sparkContext) throws DDFException {
     this.setEngineType(EngineType.SPARK);
     this.initialize(sparkContext, null);
   }
 
+  /**
+   * Use system environment variables to configure the SparkContext creation.
+   */
+  public SparkDDFManager() throws DDFException {
+    this.setEngineType(EngineType.SPARK);
+    this.initialize(null, new HashMap<String, String>());
+  }
+
+  public SparkDDFManager(Map<String, String> params) throws DDFException {
+    this.setEngineType(EngineType.SPARK);
+    this.initialize(null, params);
+  }
+
+  /**
+   * Given a String[] vector of data values along one column, try to infer what the data type should be.
+   * <p/>
+   * TODO: precompile regex
+   *
+   * @return string representing name of the type "integer", "double", "character", or "logical" The algorithm will
+   * first scan the vector to detect whether the vector contains only digits, ',' and '.', <br> if true, then it will
+   * detect whether the vector contains '.', <br> &nbsp; &nbsp; if true then the vector is double else it is integer
+   * <br> if false, then it will detect whether the vector contains only 'T' and 'F' <br> &nbsp; &nbsp; if true then the
+   * vector is logical, otherwise it is characters
+   */
+  public static String determineType(String[] vector, Boolean doPreferDouble) {
```

Comment:
This helper function should be in a util class, or `Schema`, and should probably use `ColumnType` instead of `String`.